src/context/mod.rs

```diff
@@ -37,30 +55,82 @@
 //! }
 //! ```
 //!
-//! Notice we use the [`FunctionContext`](FunctionContext) shorthand for `CallContext<JsObject>`,
-//! which is appropriate for most Neon functions.
+//! In this example, `s` is assigned a handle to a string, which ensures that the string
+//! is _kept alive_ (i.e., prevented from having its storage reclaimed by the JavaScript
+//! engine's garbage collector) for the duration of the `count_whitespace` function. This
+//! is how Neon takes advantage of Rust's type system to allow your Rust code to safely
+//! interact with JavaScript values.
 //!
-//! ### Module Contexts
+//! ### Temporary Scopes
 //!
-//! Another important context type is [`ModuleContext`](ModuleContext), which is provided
-//! to a Neon module's [`main`](crate::main) function to enable sharing Neon functions back
-//! with JavaScript:
+//! Sometimes it can be useful to limit the scope of a handle's lifetime, to allow the
+//! engine to reclaim memory sooner. This can be important when, for example, an expensive inner loop generates
+//! temporary JavaScript values that are only needed inside the loop. In these cases,
+//! the [`execute_scoped`](Context::execute_scoped) and [`compute_scoped`](Context::compute_scoped)
+//! methods allow you to create temporary contexts in order to allocate temporary
+//! handles.
+//!
+//! For example, to extract the elements of a JavaScript [iterator][iterator] from Rust,
+//! a Neon function has to work with several temporary handles on each pass through
+//! the loop:
 //!
 //! ```ignore
-//! #[neon::main]
-//! fn main(cx: ModuleContext) -> NeonResult<()> {
-//!     cx.export_function("countWhitespace", count_whitespace)?;
-//!     Ok(())
-//! }
+//! # fn iterate(mut cx: FunctionContext) -> JsResult<JsUndefined> {
+//!     let iterator = /* ... */;                  // iterator object
+//!     let next = iterator.get("next")?           // iterator's `next` method
+//!         .downcast::<JsFunction, _>(&mut cx)?;
+//!     let mut numbers = vec![];                  // results vector
+//!     let mut done = false;                      // loop controller
+//!
+//!     while !done {
+//!         done = cx.execute_scoped(|mut cx| {                   // temporary scope
+//!             let obj = next.call(&mut cx, iterator, vec![])?;  // temporary object
```

Comment:
I don't think will compile because the type `T` of `Vec<T>` is unknown.

```suggestion
//!             let obj = next.call::<_, _, JsValue>(&mut cx, iterator, vec![])?;  // temporary object
```